yfinance/utils.py

```diff
@@ -282,35 +306,48 @@ def camel2title(strings: List[str], sep: str = ' ', acronyms: Optional[List[str]
 
     # Apply str.title() to non-acronym words
     strings = [s.split(sep) for s in strings]
-    strings = [[j.title() if not j in acronyms else j for j in s] for s in strings]
+    strings = [[j.title() if not j in acronyms else j for j in s]
+               for s in strings]
     strings = [sep.join(s) for s in strings]
 
     return strings
 
 
-def _parse_user_dt(dt, exchange_tz):
-    if isinstance(dt, int):
-        # Should already be epoch, test with conversion:
-        _datetime.datetime.fromtimestamp(dt)
-    else:
-        # Convert str/date -> datetime, set tzinfo=exchange, get timestamp:
-        if isinstance(dt, str):
+def _parse_user_dt(dt, exchange_tz: Union[str, None] = None) -> Union[int, None]:
+    try:
+        if isinstance(dt, int):
+            # Should already be epoch, test with conversion:
+            dt = _datetime.datetime.fromtimestamp(dt)
+            # Convert str/date -> datetime, set tzinfo=exchange, get timestamp:
+        elif isinstance(dt, str):
             dt = _datetime.datetime.strptime(str(dt), '%Y-%m-%d')
-        if isinstance(dt, _datetime.date) and not isinstance(dt, _datetime.datetime):
+        elif isinstance(dt, _datetime.date) and not isinstance(dt, _datetime.datetime):
             dt = _datetime.datetime.combine(dt, _datetime.time(0))
-        if isinstance(dt, _datetime.datetime) and dt.tzinfo is None:
+        elif isinstance(dt, _datetime.datetime) and dt.tzinfo is None:
             # Assume user is referring to exchange's timezone
+            if exchange_tz is None:
+                exchange_tz = "UTC"
             dt = _tz.timezone(exchange_tz).localize(dt)
+
+        else:
+            raise Exception()
+
         dt = int(dt.timestamp())
-    return dt
+        return dt
+
+    except ValueError as e:
+        # Invalid date format or non-existing date (2023-20-20 or 2023-11-31)
+
+        print(f"Invalid date {dt}. {e}")
+        return None
```

Comment:
Don't discard exceptions at low-level. If one occurs it should be left to raise. Then at the API level `_get_earnings_by_date`, you can decide whether to replace with a different message.